provider/core/src/marker.rs

```diff
@@ -376,7 +376,7 @@ impl DataMarkerPath {
 
         Ok(Self {
             path,
-            hash,
+            hash: [b't', b'd', b'm', b'h', hash[0], hash[1], hash[2], hash[3]],
```

Comment:
2.46 kB seems like a lot for this commit (d7c8cedc4f5431e2778a340a9fe90476e2707efc), though I agree it should get slightly heavier since you are adding 4-byte "tdmh" tags in a bunch of places. But there shouldn't be enough keys to bloat the binary by 2.46 kB. Path tagging removed only 136 bytes.

Reply:
yeah I don't know exactly what's going on, but the tags are now on the stack instead of in static data, so that'll be more expensive

Reply:
Hmm, good observation. Is the hash field duplicated in more places, instead of being there only once in static data?

Reply:
maybe